Penteract/Assets/Scripts/AttackDroneBehavior.cpp

```diff
@@ -0,0 +1,111 @@
+#include "AttackDroneBehavior.h"
+
+#include "GameplaySystems.h"
+#include "Components/ComponentTransform.h"
+#include "Components/ComponentParticleSystem.h"
+#include "RandomNumberGenerator.h"
+
+EXPOSE_MEMBERS(AttackDroneBehavior) {
+    MEMBER(MemberType::GAME_OBJECT_UID, dronesControllerUID),
+    MEMBER(MemberType::PREFAB_RESOURCE_UID, projectilePrefabUID),
+    MEMBER(MemberType::FLOAT, droneSpeed),
+    MEMBER(MemberType::FLOAT, droneRotationSpeed),
+    MEMBER(MemberType::BOOL, droneMustHover),
+    MEMBER(MemberType::FLOAT, droneHoverAmplitude),
+    MEMBER(MemberType::BOOL, droneMustRecoil),
+    MEMBER(MemberType::FLOAT, droneRecoilOffset),
+    MEMBER(MemberType::FLOAT, droneRecoilTime),
+};
+
+GENERATE_BODY_IMPL(AttackDroneBehavior);
+
+void AttackDroneBehavior::Start() {
+    transform = GetOwner().GetComponent<ComponentTransform>();
+
+    dronesController = GameplaySystems::GetGameObject(dronesControllerUID);
+    if (dronesController) {
+        dronesControllerTransform = dronesController->GetComponent<ComponentTransform>();
+    }
+
+    hoverCurrentTime = RandomNumberGenerator::GenerateFloat(-1.5708, 1.5708);
+}
+
+void AttackDroneBehavior::Update() {
+    if (!transform || !dronesControllerTransform) return;
+
+    transform->SetGlobalPosition(float3::Lerp(transform->GetGlobalPosition(), dronesControllerTransform->GetGlobalPosition() + positionOffset + GetHoverOffset() + GetRecoilOffset(), Time::GetDeltaTime() * droneSpeed));
```

Comment:
Have you tried with slerp? Does it make it look different?

Reply:
No, I haven't. Completely forgot about slerping. I'm doing some testing now, and looks kinda buggy
![3d0920385e48f4315cdd35fe7e182ad4](https://user-images.githubusercontent.com/27434068/135923248-31211458-b64e-4a3b-825b-caec527b08fc.gif)

This is how im slerping, I haven't found any reference in our code, so I gave it a go with Normalized and Not normalized and yeah, none of them are working correctly. Will keep testing and see how it goes.
 
![image](https://user-images.githubusercontent.com/27434068/135923287-bf863a2d-e1ba-477c-9192-9faa55e5b7d2.png)

Reply:
Well, don't worry, it was just out of curiosity